### PR TITLE
Improve authors tool to include merge/commit information

### PR DESCRIPTION
**Pull request by aeslaughter**

### CODE REVIEW

**Comment by aeslaughter** [comment]
This was made to provide additional information for building the CCB membership policy.

File: python/mooseutils/tests/test_gitutils.py [hunk 1]
```diff
@@ -93,6 +93,16 @@ def testGitAuthors(self):
         with self.assertRaises(OSError) as e:
             mooseutils.git_authors('wrong')
 
+    @unittest.skip("Fails on CIVET and I can't reproduce it")
```
Comment (loganharbour) [inline python/mooseutils/tests/test_gitutils.py:96]: out of curiosity - did you try new machines?

File: python/mooseutils/tests/test_gitutils.py [hunk 2]
```diff
@@ -93,6 +93,16 @@ def testGitAuthors(self):
...
+    def testCommitters(self):
+        names = mooseutils.git_committers(mooseutils.__file__)
+        self.assertIn('Andrew E. Slaughter', names)
+        with self.assertRaises(OSError) as e:
+            mooseutils.git_authors('wrong')
+
+        names = mooseutils.git_committers(mooseutils.__file__, '--merges')
+        self.assertIn('Logan Harbour', names)
```
Comment (loganharbour) [inline python/mooseutils/tests/test_gitutils.py:104]: 👋🏼

**Review by loganharbour — APPROVED** [review]

File: python/mooseutils/tests/test_gitutils.py (hunk 1 above)
Comment (aeslaughter) [reply python/mooseutils/tests/test_gitutils.py:96]: I have not.

**Pull request merged**